builder/job.go

```diff
@@ -142,6 +143,16 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 
 	sf := streamformatter.NewStreamFormatter(buildConfig.JSONFormat)
 
+	var defaultArchiver *archive.Archiver
+	uidMaps, gidMaps := d.GetUidGidMaps()
+	if uidMaps != nil || gidMaps != nil {
+		defaultArchiver = &archive.Archiver{
```

Comment:
Isn't it better to set `UidMaps` and `GidMaps` for `archive.Archiver{Untar: chrootarchive.Untar}` if they're needed? Then you can use just `archiver` and remove a lot of checks for nilness.